pkg/collector/corechecks/containerlifecycle/check.go

```diff
@@ -122,9 +122,17 @@ func (c *Check) Run() error {
 
 	for {
 		select {
-		case eventBundle := <-contEventsCh:
+		case eventBundle, ok := <-contEventsCh:
+			if !ok {
+				stopProcessor()
```

Comment:
same here, it seems possible to do `defer stopProcessor()` outside the `for {}` loop.